pex/environment.py

```diff
@@ -192,6 +193,18 @@ def _activate(self):
 
         if os.path.isdir(dist.location):
           with TRACER.timed('Adding sitedir', V=2):
+            if dist.location not in sys.path and self._inherit_path == "fallback":
+              # Prepend location to sys.path.
+              # This ensures that bundled versions of libraries will be used before system-installed
+              # versions, in case something is installed in both, helping to favor hermeticity in
+              # the case of non-hermetic PEX files (i.e. those with inherit_path=True).
+              #
+              # If the path is not already in sys.path, site.addsitedir will append (not prepend)
+              # the path to sys.path. But if the path is already in sys.path, site.addsitedir will
+              # leave sys.path unmodified, but will do everything else it would do. This is not part
+              # of its advertised contract (which is very vague), but has been verified to be the
+              # case by inspecting its source for both cpython 2.7 and cpython 3.7.
+              sys.path = [dist.location] + sys.path
```

Comment:
maybe mutate vs replace here in case anything else has an existing ref to the original `sys.path`?

```
sys.path.insert(0, dist.location)
```

this is more in-line with `addsitedir`'s `sys.path.append()` too.

Reply:
Done